### PR TITLE
Refactor and fix Search

### PR DESCRIPTION
**Pull request by jbydeley-d2l**
- Separate out pieces of App.vue into new components
- Use CSS in their own components
- Fix search so it doesn't close the dialog when searching

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -79,56 +55,36 @@
...
-      searchBy: ''
-    }
+  components: {
+    EmptyMessage,
+    UserSearch,
+    ExemptButtonGroup
   },
 
   created() {
+    console.log('Reloaded')
```
Comment (RyanSweetD2L) [inline src/App.vue]: Remove console.log

**Comment by RyanSweetD2L** [comment]
:shipit:

File: src/App.vue [hunk 2]
```diff
@@ -1,54 +1,30 @@
...
-          spellcheck="false"
-          autocomplete="off">
-        <button v-if="showSearchButton(searchBy)" :aria-label="$t('ariaSearchButton')" class="vui-input-search-button" @click="searchUsers(searchBy)"></button>
-        <button v-else-if="!showNoUsers" :aria-label="$t('btnClearSearch')" class="vui-input-search-clear-button" @click="clearResults"></button>
-      </form>
-    </div>
+    <exempt-button-group
+      :show-exempt="!showNoUsers" />
+
+    <user-search />
```
Comment (lawong-d2l) [inline src/App.vue]: Everything looks a lot cleaner! Is Ed okay with the new UI element placements?

Comment (jbydeley-d2l) [reply src/App.vue]: Did the UI change? It should look the same as before.

Comment (lawong-d2l) [reply src/App.vue]: Yah it's a tiny thing. I don't think it's a big deal. The search bar was on the same line as the exemption button group. The clear, was on the same line as exemptions count.

**Pull request merged**